### DISCUSSION TITLE
[BRAINSTORMING] Ideas the TAG could focus on in 2024 + Kick off Roadmap planning

### DISCUSSION THREAD

**Discussion by leonardpahlke** [cncf/tag-env-sustainability#277] [General]
⚠️ **DISCUSSION CLOSED - PROPOSALS ARE IN DISCUSSION https://github.com/cncf/tag-env-sustainability/issues/321 - IF YOU LIKE YOU CAN STILL ADD (SUB) COMMENTS TO COMMENTS** ⚠️

There are sooo many ideas out there what the TAG ENV could work on. A few of them are already codified in issues (maturity model, cloud native sustainability learning path, kubernetes sustainability whitepaper, etc.) or even completed or in the work (green project reviews, cloud native sustainability week, landscape document, etc.).

Let's collect ideas we all have to push environmental sustainability in the TAG and then decide together which ideas we like to work on in 2024.

1. Brainstorm ideas and translate them into issues (in this discussion)
2. Roadmap planning, discuss and define as a group which ones to work on until [Kubecon EU (mid march)](https://events.linuxfoundation.org/kubecon-cloudnativecon-europe/) (in a follow up issue)

Notes: I would like to make sure that the roadmap captures technical and non-technical topics to enable as many potential contributors as possible. With WG Comms and WG Green Reviews, we already have two projects locked in. Assessing the capacity is a question we need to ask ourselves, ad-hoc 1x Chair : 1x project, WG are separate, if we find more leads could extend the capacity. 

**Anyone can comment here and share their thoughts! 😁🙌**

**Comment by brianirish** [comment]
**Context:**
In the United States, the US Green Building Council ([USGBC](https://www.usgbc.org/)) has a program called LEED (Leadership in Energy and Environmental Design) specifically to [rate buildings with a sustainability score](https://www.usgbc.org/leed). This program has been a huge success, and many construction companies in the US share with pride that they are LEED certified.

**Proposal:**
There are many parallels to draw between what the USGBC has done for sustainability in the construction industry, and what this TAG is chartered to do for the computing industry. As a group, we can and should consider drawing on the success of the LEED program, and come up with a similar program of our own. A "System Energy and Environmental Design" (SEED) Program is my most obvious idea for a name for this program, which would also lend itself well to a logo (the seed of a plant, green, etc.) and provide similarity to the LEED program so that businesses/projects that are based in the US would understand it faster.

As there are with LEED, there would be varying levels of SEED certification based on how sustainable you are as a company. Determining the metrics for this would be a challenge, which leads me to the next section:

**Challenges:**
- Having the software in place to accurately (to a point) measure the things that are relevant for calculating a score based on the [SCI specification](https://greensoftware.foundation/articles/software-carbon-intensity-sci-specification-project).
- Determining the definitions and thresholds for each level of SEED rating.

Thanks! I really welcome feedback. I'm new to the group, so I apologize if this has been discussed before.

**Reply by JacobValdemar** [reply]
This makes me think of the [EU Energy Label](https://en.wikipedia.org/wiki/European_Union_energy_label). I am not familiar with the LEED program. But giving a rating could make sense to guide decisions when choosing between many similar projects that does the same. E.g. when looking at the CNCF Landscape and thinking Flux vs Argo.
![New_European_Union_energy_label_from_2021](https://github.com/cncf/tag-env-sustainability/assets/16527874/85fc041c-3d8c-4ffb-a298-22c0f8a5fc29)

**Reply by josephnatale** [reply]
@JacobValdemar - I like where this is going, could be a badge, tag, or something else application strive to achieve.  I do worry a bit about how non-functionals like availability, RTO, RPO, etc will have an impact on this.  IE - a highly critical app that needs to run active/active across 2 geo-locations vs something non-critical running out of a single location, the requirements of the highly critical application will drive up its carbon usage.  

It might make sense to break this down into:
1. total carbon emissions based off SCI spec or something else
2. CNCF sustainability efficiency rating (figure out the naming convention later 😄 ) takes into account some kind of 'complexity factor'

**Reply by leonardpahlke** [reply]
A good idea. We discussed this in the TAG some time ago, but have not yet launched a project. There are several other initiatives in this direction. This is another one https://www.blauer-engel.de/en/productworld/resources-and-energy-efficient-software-products. I know that GSF is also working on this topic. This is along the lines of the work that the wg-green-reviews is currently doing.

**Reply by leonardpahlke** [reply]
I like the proposal. It is definitely of interest to us in the TAG. The scope of the proposal to define a badge that can be used to certify the sustainability footprint of software is out of scope.

1. we focus on Cloud Native Sustainability. These badges target a broader scope that is better captured by the Green Software Foundation for example.
2. developing certifications and badges for projects is a big undertaking. It is a lot bigger than the WG Green Reviews. We don't have full-time employees or generally don't have the resources for something like this.
3. There are also concerns about "green washing". In the WG Green Reviews we can put this to rest to some extent as all CNCF projects are open and transparent as required by the Code of Conduct. In a larger scope this is not the case.
3. A project like this takes us a bit in the direction of providing services (certification services) that we should not provide according to our charter.

It would be good to collaborate or stay in touch with another organization that deals with this issue.

**Reply by leonardpahlke** [reply]
Thanks for proposing the idea! @brianirish

**Reply by brianirish** [reply]
All fair points, @leonardpahlke. Thanks for taking the time to walk through it.

**Comment by ArneTR** [comment]
As requested by @leonardpahlke : Cross-referencing the idea from https://github.com/cncf/tag-env-sustainability/pull/283#issuecomment-1819639008 to benchmark the reduced network and maybe compute cost from using Buildpack container rebases than having full docker rebuilds in a reference CNCF project that has migrated to the technology.

Referencing a post regarding the general approach suggested here: https://github.com/cncf/tag-env-sustainability/discussions/273

**Comment by felipehuici** [comment]
Hi all,

This is somewhat related to Arne's comment, but more generally it'd be good to have a topic on the efficiency of deployments, including, but not limited to, KPIs such as:

* Image size
* Build times
* Memory consumption
* Server density, ie, how many pods/instances/containers can run on a single server
* Cold boots (because slow boots often lead to over-provisioning)
* Performance (eg, reqs/sec for a web server), which goes to efficiency/power consumption

An advantage here is that it'd be "easy" to measure/compare efficiency via the metrics above. I'm relatively new to this group too, so any feedback is really appreciated. 

Full disclosure: I have a horse in this race in the form of unikernels (and more specifically the OSS Unikraft project), highly specialized VMs that can reduce the metrics above significantly (e.g., an OCI-compliant NGINX image of just 1.25MBs, cold boots of a few millis, memory consumption of a few MBs, etc).

**Reply by wrkode** [reply]
This could become a byproduct of the WG Green Review effort.

**Reply by leonardpahlke** [reply]
@ArneTR @felipehuici thanks for the idea (this comment + @ArneTR https://github.com/cncf/tag-env-sustainability/discussions/277#discussioncomment-7627032). This falls within the scope of TAG and is feasible to a certain extent. There several KPIs mentioned that we need to see which are realistic to capture. Maybe just focusing on one, like the networking part could already suffice. Depending on which ones we choose we need to see if folks with expertise are interested to support.

Yes, @wrkode this is related to the work of the WG Green Reviews, but from my point of view is not possible for the WG this year as it already has a full agenda. Its also not desirable as this can be captured as an independent project and would just disturb the WG imo. We could implement this proposal in one or more whitepaper(s). Whether we can schedule this as a goal for this year depends somewhat on the community's interest in helping with this, as written. However, it looks like there are several people interested in helping so i am hopeful!

@ArneTR @felipehuici what do you think about the idea of writing a whitepaper on this proposal - with a limited scope, perhaps on the networking part. Are either or both of you interested in leading this initiative with other TAG members?

Thanks for the proposal!

**Reply by ArneTR** [reply]
@leonardpahlke I do like the idea of a whitepaper, however I would not favor of only restricting it to Network Transfer.
Although this value is most likely the one with the highest emissions it is also the most volatile. 
It is based on so many assumptions that people might see it as not actionable or too vague.

Numbers like operational energy / carbon and also embodied carbon are far more concrete and are often measureable (if we use our infra to benchmark for instance) or require far less assumptions if we use AWS / GCP etc. products.

Since the tools are there already I would argue not to restrict the scope.

**Reply by leonardpahlke** [reply]
Okay. we should not limit the scope without any reason of course. If there are low hanging fruits we should collect them! Right, networking is the biggest question mark, thats why I was proposing to focus just on that. However, the other data would be very helpful too. So this can be shaped out either way, i think its good and gives value. The direction you propose sounds good to me!

**Comment by mbianchidev** [comment]
Is there a plan to create a sustainability card in the main CNCF Landscape? 

It would be cool to showcase a catalog of all the tools that help measuring and/or reducing the carbon footprint of Cloud Native workloads, basically a condensed version of [this landscape document](https://tag-env-sustainability.cncf.io/landscape/) in the style of the [serverless one](https://landscape.cncf.io/serverless)

**Reply by leonardpahlke** [reply]
@mbianchidev thanks for proposing the idea. This is a topic we discussed in the past. There is currently a rewrite of the landscape document underway see https://github.com/cncf/tag-env-sustainability/issues/302 (cc @by-d-sign @guidemetothemoon). Depending on the outcome this could be a good followup action. I could see it shape out in similar style as we did with the WASM landscape https://landscape.cncf.io/?group=wasm - a separate tab with custom categories.

**Comment by SamYuan1990** [comment]
Context: 
Considering sustainable is a global action, hence we will need i18n to avoid language barrier for audience.

Proposal:
- i18n and different timezone meeting.
- maybe we should considering have a keynote on kubecon China/Asia in 2024

**Reply by musingvirtual** [reply]
I strongly support this, APAC contributors often get left out!

**Reply by leonardpahlke** [reply]
Thanks for this suggestion @SamYuan1990. We are planning to start monthly Asia-friendly meetups that are either Europe + Asia or NA + Asia-friendly. Over the past year, the TAG has greatly expanded its contributor base in the Asian regions and there is a need to organize these meetups! As for Kubecon China, as you know, we had a meeting. We will see what we can do.

**Comment by SamYuan1990** [comment]
Context:
Step further with landspace document.
@felipehuici's comments give me a this inspiration.
Ref to previously work in Chinese as I searched for a web edition as https://baijiahao.baidu.com/s?id=1766556828718216261&wfr=spider&for=pc

Briefly:
in this blue paper, we made a cube model for sustainable computing and each surface is a point of view.
For an example, 
- The front surface sustainable computing is full stack from data center to application. 
- The top surface sustainable means we need to balance energy with performance, security and etc....
We also go through many existing metrics as PUE and others. 

Proposal:
For example as [Peaks](https://github.com/sustainable-computing-io/peaks)(cc @rootfs, @marceloamaral and @husky-parul here) as kepler community is planning for schedule k8s pod not only by hardware resources but also energy consumption by re-scoring the pod among k8s nodes.
Hence, maybe we can build up some metrics and peaks is an example project which implements that metrics?
Basing on that implementation(or best practices?), we can update blog and step further with our landspace document?

p.s.  I will update the English version once I got it, and the original link seems not available for open today. Hence I am sorry for a 3rd party web edition...

**Reply by leonardpahlke** [reply]
I hope i understood correct, its about collecting metrics? Building up some sort of database for projects like kepler to use - to train ML models etc? Am I correct, could you @SamYuan1990 expand if i missunderstood? Thanks!

**Comment by nikimanoledaki** [comment]
In 2024, it would help tremendously to continue gathering and documenting user stories for tools and patterns discussed in TAG ENV.

This is less of a concrete initiative and more of a general reflection of what we should continue to focus on in our TAG strategy. We are already doing a lot of this! I would like to highlight what we should _continue to do_ / _do more of_ - general things that help with adoption. Let’s continue to encourage this open forum where people can share how to implement solutions on environmental sustainability using cloud-native tooling! :)

### What?

Let's gather user stories and collect data on carbon savings from the CNCF ecosystem, the various TAG community members, End User group, GSF, other TAGs, etc.

We are doing this in various initiatives such as the landscape, Cloud Native Sustainability Week, around KubeCon + CloudNativeCon, etc (apologies if I am missing other important milestones here!).

Specifically, it would be great to continue gathering examples of:
- Data on carbon, energy, and cost savings that were achieved.
- How cloud-native tools were implemented to optimise for energy/carbon monitoring and reduction.
- How tooling specifically for energy/carbon reduction were used in a cloud-native environment.

Going forward, we could make a call for content that showcases user stories at various stages: 
- Experimental: POCs, demos, lessons learned, etc. Tell us what worked or didn't work.
- Platform: Are you running something in a dev environment or even prod?! Tell us about it!

### How?
- Social media strategy
  - We now have WG Comms! I am not so active there due to time constraints but it is so great to see this WG growing 💚
- Blog
  - For the content itself, it depends on whether folks prefer a written or presentation format. We have shared a few blog posts already. Maybe we could do a specific call for content around user stories and data around carbon-savings?
- TAG meetings
  - For presentations, we could lend time during the TAG meetings themselves so that folks can showcase demos or do small presentations. We can reuse some of the process that we have for doing demos but this needs some improvement. A Google form might be better so that we can have a drop-down list with future TAG meetings to choose from.
  - This may be too far-fetched of an idea but, since the videos are uploaded to YouTube, maybe we could stop the recording after the demo and start it again afterward once the meeting starts so that 2 separate videos are uploaded. Just an idea 😄

**Reply by musingvirtual** [reply]
This is a solid idea and I'm happy to talk to you about how we collect end user stories in OpenTelemetry, since you can apply some of the tactics we used.

**Reply by nikimanoledaki** [reply]
### Suggested actions (TAG meeting 20/12/2023)

- New format for demos in regular TAG meetings where we highlight wins and/or blockers (as mentioned by @wrkode) for implementing solutions for environmental sustainability in cloud-native
- [@leonardpahlke] Target End User group in CNCF
- Interviews at KubeCon Europe
- [@guidemetothemoon] Communication plan in WG Comms to **specifically call out** that we would like to collaborate on blog posts about this
- [Niki] Similar to GSF blog posts with interviews/use cases

**Reply by nikimanoledaki** [reply]
@musingvirtual your input would be greatly appreciated! This would fit really well with the WG Comms and could help shapre the WG's strategy. Could you share some of your top tips in the CNCF Slack channel [#tag-env-wg-comms](https://cloud-native.slack.com/archives/C068XUD9AEA) please :)

**Reply by leonardpahlke** [reply]
Thanks for the proposal Niki! I think this is a ongoing process where we need to think about how to improve communications and how to interact with other community groups. I hope the idea of TAG Ambassadors could help https://github.com/cncf/toc/issues/1211. WG Comms can take the ideas proposed as input to improve communication, outreach etc. cc @caradelia @patricia-cahill

**Reply by vallss** [reply]
As I wrote in the Slack channel, I would really like to help with this. I was already trying to gather information, data and some case studies/use cases that could be used as a starting point for the company I work for (also, I myself am very much interested in this).

**Comment by mbianchidev** [comment]
Have Kubetrain as official project for the TAG, with a WG and all? It's a pilot project for KubeCon Paris 2024 but we might want to extend it to KubeCon NA 2024 as well with San Francisco - Salt Lake route and a couple of other routes.

If these two pilots succeed it could be possible making it recurrent every year, hence it should (imho) be an official TAG project.

CC: @AndreaGiardini

**Reply by musingvirtual** [reply]
There are also usually a handful of people who cycle long distances to Kubecon and post about their tour on social media to highlight, although I don't think it's an organized group!

**Reply by musingvirtual** [reply]
Another conference-related option that might be nice if we can think up something is to do some sort of environmental work & discussion day before or after the event in the city that Kubecon is in - that would get people out and moving, a workday with natural light and some physical tasks would be nice after a long flight and help your body adjust to the new timezone, and we can of course talk about the TAG.

**Reply by leonardpahlke** [reply]
Thanks for the suggestion, @mbianchidev! Kubetrain is indeed a great initiative. However, it's important to remember that our focus here is on technical advice for cloud native sustainability (TAG = Technical Advisory Group), not activism. It is critical that we stick to this mission and we need to avoid distractions. We have more ideas to explore than we have capacity for, and while Kubetrain is a substantial project requiring significant resources, its scope and scale make it unsuitable to tackle. I am sure TAG members are already and will continue to help and organize, Kubetrain cannot be a sub-project or working group under the TAG umbrella for the reasons mentioned above.

**Comment by wrkode** [comment]
Adding here my Slack post:

Hello people,
I've been doing some thinking on the cluster scope and target at #wg-green-reviews.
We're now focusing on the power consumption and utilization. Which makes perfect sense.
I've been thinking on how we could expand the effort in CO2 emission calculation and Power Usage Efficiency.
CO2 emissions are calculated as E*CO2i (energy utilized * Carbon Intensity).
The generally accepted issue with this calculation is the location, which directly affects the Carbon Intensity.
in other words, 1KWh in London will produce a (obviously) different Carbon footprint than a 1KWh Oslo. On top of this, Hyperscalers do not publish (near) real-time data on the footprint of services and workload, for a given service in a given region. All data are 6 to 12 months old.
We can’t directly influence the issue above, but we can work through it, until the Hyperscalers will provide us with real-time or near-real-time data.
I've been fiddling with the idea of a "Rosetta Stone for CO2 intensity by region".
The idea would be to have a region aware/bound intensity array that is read by tooling with access to CPU/GPU/TPU type. I'm coding a PoC for this.
meanwhile, I found the work of these guys, quite interesting and very close to what I have in mind.
https://calculator.green-algorithms.org/
I'd like to have some opinions and reality checking by the TAG contributors and leads. Do you believe this falls into #wg-green-reviews? is this something or am I navigating in star wars hyperspace?
I've seen several initiatives and concepts aiming to reduce the CO2 footprint of software and platform.
The majority is addressing the region where the workload is deployed, moving workloads into "greener" regions rather than optimizing code, the platform and the proximity to similar or invasive workloads.
I find this approach very limited and (sorry to say) lazy from an engineering perspective.

**Reply by rossf7** [reply]
Hi @wrkode, I like this idea a lot and I think we've been hanging out in similar patches of hyperspace ;)

For the carbon intensity part I've been thinking of something similar.

I'd like to see an exporter that finds out which cloud and region its running in from node metadata and exposes the current and forecast carbon intensity using APIs like WattTime and Electricity Maps.

The carbon-intensity-exporter developed by Microsoft for the carbon-aware-keda-operator can do that for Azure if you configure the region manually.

https://github.com/Azure/kubernetes-carbon-intensity-exporter
https://github.com/Azure/carbon-aware-keda-operator

It uses the GSF Carbon Aware SDK which has the lat / lon coordinates that Microsoft publishes for all Azure regions. Both WattTime and Electricity Maps have API endpoints to get the grid region by geolocation.

I've worked on a similar tool for Green Web Foundation but it doesn't have the region to grid mapping yet.
https://github.com/thegreenwebfoundation/grid-intensity-go

For Impact Framework hopefully there will be a model for the cloud regions with geolocation that adds the AWS and GCP regions. Similar to the cloud instance metadata model.

https://github.com/Green-Software-Foundation/real-time-cloud/issues/7#issuecomment-1872622694
https://github.com/Green-Software-Foundation/if-models/tree/main/src/lib/cloud-instance-metadata

This model could be used as the source of truth for the "Rosetta stone tool".

The Real Time Cloud project is also planning to package the PUE and CFE datasets published by the cloud providers as IF models. This is more challenging because of gaps and discrepencies between providers in the data. However they could also be used as inputs to the tool. 

For WG Green Reviews I'd say this tool is outside of scope for now. Our initial focus is on providing sustainability metrics to CNCF projects and helping them optimise to consume less resources.

But I think carbon intensity is important to the TAG. So maybe we could start a project to build a PoC of this tool? Or through the TAG gather a group of people that would like to work on this?

William, the scope I've been thinking of is quite narrow around carbon intensity. Are you thinking of a wider scope? 

I totally agree we need real time metrics for energy consumption and carbon intensity but this tool could bridge the gap until they are available.

**Reply by leonardpahlke** [reply]
@wrkode this is outside the scope for the day, if I have understood it correctly. It sounds like you are thinking about developing a open source project / product and the TAG does not do that. I would follow it up, sounds like you may be on to something but its a bit early to tell, would be interesting to read a more thorough project description. Maybe folks from the TAG are interested in supporting the development outside of the TAG.

**Reply by wrkode** [reply]
@rossf7 thanks for the interest

@leonardpahlke, ok, I'll go on with developing the Project Description and PoC. will come back to the TAG for further discussion once I have more reading.

**Comment by sergioarmgpl** [comment]
I am exploring projects around food sustainability.

Let’s say that I will try to implement an architecture diagram for farmers maybe to take advantage of it, I will use speaking sessions on events to push myself to develop the idea. My outcome will bring like diagram of project ideas that Farmers can implement. @leonardpahlke well, looking to give some env sustainability architectures.

**Reply by leonardpahlke** [reply]
Hey @sergioarmgpl thanks for the suggestion. It is important to remember that our focus here is on technical advice for cloud native sustainability (TAG = Technical Advisory Group) and food sustainability is out of scope. We have more ideas to explore than we have capacity for, and while food sustainability is an interesting area to explore, its scope makes it unsuitable to tackle for us.

@sergioarmgpl did you saw the UN Sustainability Goals Hackathon at Kubecon Paris? Sounds like your idea goes in this direction ;) https://www.cncf.io/blog/2023/12/20/kubecon-cloudnativecon-europe-hackathon-challenges-brought-to-you-by-the-united-nations/

**Comment by musingvirtual** [comment]
I'm from the OpenTelemetry project.  There seems to be interest in engaging us and I agree that telemetry is key to sustainability, but we need to have some dialogue and education between the two groups.  It might also be good for someone to engage tag-observability as a whole.  If you are further along with sustainability work in another project like Prometheus, we can possibly set up ways to ingest their signals as a quick option.

Sustainability is a question with a large cross-cutting scope and it's important that we actually have a thoughtful approach, it's not about one or two added metrics.  To add sustainability to our spec is not just about defining a list of metrics, we'd need to make some decisions within the technical committee & the spec working group about where sustainability metrics would sit in our spec and also define how traces and logs and other signals would be handled, then there is project management to enable our working groups to implement the metrics once they are in spec, there is documentation, and finally we need to educate OpenTelemetry adopters so the metrics get used.

This is a lot of work for contributors who are already stretched thin and we need help from your group if we're going to implement it.  In order to make this a priority we'd need to recruit and train some humans who can be dedicated to shepherding this along through the project.

Here are some specific ideas for collaboration:

My end user working group which organizes events can commit to hosting a speaker or panel from tag-sustainability if you provide them to frame the problem space.  We can also commit to hosting ad-hoc follow up discussion sessions between TAG-Sustainability and OpenTelemetry until we decide whether to charter a working group or move forward in another way.  

I can also commit to using our communication channels through the OpenTelemetry comms working group and asking observability experts to post about ongoing work, in order to try to locate people within the observability community who are interested in the problem. 

Austin Parker and Ed Vielmetti are exploring a collaboration on how we might use existing OpenTelemetry to do sustainability work using the Equinix data center and Honeycomb, which will help us kick off education and awareness efforts with something we can implement right away.

OpenTelemetry has a lounge space on the KubeCon EU show floor and we can also do programming together in person there, and while the Observability Day program is being set right now we may be able to squeeze in a lightning talk or something.

Can TAG-Sustainability commit to hosting a talk from OpenTelemetry about the basics of how to collaborate with us and the basics of OpenTelemetry, so that you can help us resource this kind of work appropriately?  And can your comms working group commit to using your channels to help us find contributors, and to cross-promote the discussions I'm outlining?

**Reply by nikimanoledaki** [reply]
Thank you @musingvirtual - added a cross-reference to your suggestions in this issue to share with the other group & track there as well: https://github.com/open-telemetry/opentelemetry-specification/issues/3345

**Reply by catblade** [reply]
Let's get you on the schedule.  Please reach out to any chair or tech lead to get this on.

**Reply by nikimanoledaki** [reply]
📆 -wise, we have a demo scheduled for the next regular TAG meeting on January 31st so the next available date would be Wednesday February 14th

**Reply by nikimanoledaki** [reply]
@musingvirtual hi! Would you be able to join the regular TAG ENV meeting on the 14th of February to discuss this?
Calendar link [here](https://calendar.google.com/calendar/u/0/embed?src=72e93a411f02e5664bb4485c04311b83dae6a62574e4ab882a1ccf8526aa9bf1@group.calendar.google.com&ctz=America/Chicago)

**Comment by octonawish-akcodes** [comment]
**Expanding on the Newsletter Idea for Our TAG:**

I had the idea of integrating a newsletter into the website, especially considering the impactful work the TAG does within the open-source community. This newsletter could serve as a dynamic platform to align with the mission of focusing on environmental sustainability in the cloud-native landscape.

**Newsletter Theme and Focus:**
I think each edition of the newsletter can highlight key factors influencing environmental sustainability in the cloud-native ecosystem. We can feature open source projects from our TAG that are actively working to reduce carbon footprints and share updates on initiatives aimed at optimizing and minimizing the environmental impact of cloud-native infrastructure.

**Sharing Educational Content:**
We can use this newsletter which contains articles, case studies, and best practices that can be curated to help our community better understand and contribute to sustainable practices.

**Project Spotlights:**
We can also feature regular spotlights on these projects, sharing their progress, achievements, and ways our community can actively contribute or get involved.

**Community Engagement like surveys, or polling:**
Interactive elements in the newsletter, like polls or surveys, can help foster engagement. We want to encourage our community members to share their insights, experiences, and ideas about environmental sustainability in the cloud-native space.

**Event Announcements:**
We can also use this newsletter to announce and promote events, webinars, or conferences related to environmental sustainability and cloud-native development. This could include both TAG-specific events and other industry conferences that align with our mission.

**Metrics and Impact Measurement:**
Regularly sharing metrics related to the impact of our open-source projects on reducing carbon footprint is essential. We want to demonstrate the positive changes we're making as a community and how our efforts contribute to broader sustainability goals.

I believe that by implementing a newsletter with these tailored features, we can effectively leverage this communication channel to raise awareness, educate our community, showcase projects, and actively contribute to positive environmental change within the cloud-native landscape.

Relevant discussion: https://cloud-native.slack.com/archives/C03F270PDU6/p1702403586261379

**Reply by leonardpahlke** [reply]
@caradelia / @patricia-cahill  / folks from WG Comms could you take a look at this one and share your thoughts. Thanks!

**Comment by xamebax** [comment]
👋🏼  I'd like to pick up the work on [best practices for running sustainable Kubernetes clusters](https://github.com/cncf/tag-env-sustainability/issues/54). 

To quote from the original issue:

> Kubernetes is the key driver and foundation of the cloud native universe. Within this project, we will work on the basic capabilities, best practices and scheduling mechanism as well as extensions to improve the K8s GHG impact. 

> Where are patterns, there are anti-patterns, where we improve for speed, resistance and security, there is space for configurations and setting with a negative impact on power consumption and the following chain.

**Reply by JacobValdemar** [reply]
I would also like to help with this. I have time after January 2024. I have learned a lot from writing my master's thesis _Carbon Efficient Karpenter: Optimizing Kubernetes Cluster Autoscaling for Carbon Efficiency_ which I would love to share 💚

**Reply by leonardpahlke** [reply]
Thanks for bringing this idea up again @xamebax! We already planned for this in the past but as you saw, we didnt had the capacity back then. It more achivable this year. @xamebax / @JacobValdemar are either or both of you interested in leading (/ co-leading) this project with support from other TAG members?

**Reply by JacobValdemar** [reply]
Yes, I can lead this project, and if @xamebax is up for it, we can co-lead it together

**Reply by leonardpahlke** [reply]
Awesome!

**Reply by xamebax** [reply]
Yes, I would love to co-lead this! 🙂
